src/physics/reynolds_averaged_navier_stokes.cpp

```diff
@@ -0,0 +1,823 @@
+                    jac_grad_row += jacobian_wrt_gradient[d_gradient][sr][sc]*manufactured_solution_hessian[sc][d_gradient][d]; // symmetric so d indexing works both ways
+                }
+            }
+            dissipative_flux_divergence[sr] += jac_grad_row;
+        }
+    }
+    std::array<real,nstate> dissipative_source_term;
+    for (int s=0; s<nstate; s++) {
+        dissipative_source_term[s] = dissipative_flux_divergence[s];
+    }
+
+    return dissipative_source_term;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+std::array<real,nstate> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::physical_source_source_term (
+    const dealii::Point<dim,real> &pos,
+    const dealii::types::global_dof_index cell_index) const
+{    
+    // Get Manufactured Solution values
+    const std::array<real,nstate> manufactured_solution = get_manufactured_solution_value(pos); // from Euler
+    
+    // Get Manufactured Solution gradient
+    const std::array<dealii::Tensor<1,dim,real>,nstate> manufactured_solution_gradient = get_manufactured_solution_gradient(pos); // from Euler
+    
+    std::array<real,nstate> physical_source_source_term;
+    for (int i=0;i<nstate;i++){
+        physical_source_source_term = physical_source_term(pos, manufactured_solution, manufactured_solution_gradient, cell_index);
+    }
+
+    return physical_source_source_term;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+void ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::boundary_manufactured_solution (
+    const dealii::Point<dim, real> &pos,
+    const dealii::Tensor<1,dim,real> &normal_int,
+    const std::array<real,nstate> &soln_int,
+    const std::array<dealii::Tensor<1,dim,real>,nstate> &soln_grad_int,
+    std::array<real,nstate> &soln_bc,
+    std::array<dealii::Tensor<1,dim,real>,nstate> &soln_grad_bc) const
+{
+    // Manufactured solution
+    std::array<real,nstate> conservative_boundary_values;
+    std::array<dealii::Tensor<1,dim,real>,nstate> boundary_gradients;
+    for (int s=0; s<nstate; s++) {
+        conservative_boundary_values[s] = this->manufactured_solution_function->value (pos, s);
+        boundary_gradients[s] = this->manufactured_solution_function->gradient (pos, s);
+    }
+
+    for (int istate=dim+2; istate<nstate; ++istate) {
+
+        std::array<real,nstate> characteristic_dot_n = convective_eigenvalues(conservative_boundary_values, normal_int);
+        const bool inflow = (characteristic_dot_n[istate] <= 0.);
+
+        if (inflow) { // Dirichlet boundary condition
+            soln_bc[istate] = conservative_boundary_values[istate];
+            soln_grad_bc[istate] = soln_grad_int[istate];
+        } else { // Neumann boundary condition
+            soln_bc[istate] = soln_int[istate];
+            soln_grad_bc[istate] = soln_grad_int[istate];
+        }
```

Comment:
Does it work when its consistent with `navier_stokes.cpp` rather than `euler.cpp` ?

Reply:
fixed